Userland/Libraries/LibAccelGfx/Painter.cpp

```diff
@@ -190,16 +195,37 @@ void Painter::draw_scaled_bitmap(Gfx::FloatRect const& dst_rect, Gfx::Bitmap con
     glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, scaling_mode_gl);
     glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, scaling_mode_gl);
 
-    auto vertices = rect_to_vertices(to_clip_space(transform().map(dst_rect)));
-    auto texture_coordinates = rect_to_vertices(to_texture_space(src_rect, bitmap.size()));
+    auto dst_rect_in_clip_space = to_clip_space(transform().map(dst_rect));
+    auto src_rect_in_texture_space = to_texture_space(src_rect, bitmap.size());
+
+    Vector<GLfloat> vertices;
```

Comment:
Can you `ensure_capacity` here, or even better use a more static container?

Reply:
makes sense, I added `ensure_capacity`.